content/sensu-core/2.0/reference/entities.md

```diff
@@ -4,82 +4,355 @@ description: "The entities reference guide."
+## New and improved entities
 
-## Agent entities
+An `entity`, formally known as a `client` in Sensu 1.x, represents anything (e.g. server, container, network switch, etc) that needs to be monitored. Sensu 2.0 uses an updated data model that allows for it to acknowledge the full range of infrastructure, runtime and application types that compose a complete monitoring environment (from server hardware to "serverless" functions). Sensu no longer focuses on the object doing the monitoring and instead focuses on the object it monitors. We call these monitored parts of an infrastructure "entities." An Entity not only provides context to event data (what/where the event is from) but an event's uniqueness is determined by the check name and the ID of the entity upon which the check ran. In addition, an entity can contain system information such as the hostname, OS, platform, and version. 
 
-### What is an agent entity?
+## Entities specification
 
-Agent entities are monitoring agents, which are installed and run on every system that needs to be monitored. The entity is responsible for registering the system with the Sensu backend service, sending keepalive messages (the Sensu heartbeat mechanism), and executing monitoring checks. Each entity is a member of one or more **subscriptions** – a list of roles and/or responsibilities assigned to the agent entity (e.g. a webserver, database, etc). Sensu entities will "subscribe" to (or watch for) check requests published by the Sensu server (via the Sensu Transport), execute the corresponding requests locally, and publish the results of the check back to the transport (to be processed by a Sensu server).
+### Entity Attributes
 
-### Creating an agent entity
+ID           | 
+-------------|------ 
+description  | The unique ID of the entity, validated with go regex [`\A[\w\.\-]+\z`](https://regex101.com/r/zo9mQU/2)
+required     | true
+type         | string 
+example      | {{< highlight shell >}}"ID": "example-hostname"{{</ highlight >}}
 
-To create an agent entity, `sensu-agent` must be installed. Upon spinning up the sensu-agent, you can use several flags to customize the properties of the entity. In the example below, the optional flags are used to describe the default values of the agent entity that gets created upon starting sensu-agent. Other flags/properties you can set for your agent entity include: `--deregister`, `--subscriptions`, `--deregistration-handler`, and `--redact`.
+class        | 
+-------------|------ 
+description  | The entity type, validated with go regex [`\A[\w\.\-]+\z`](https://regex101.com/r/zo9mQU/2)
+required     | true
+type         | string 
+example      | {{< highlight shell >}}"class": "agent"{{</ highlight >}}
```

Comment:
We should probably list the allowed values, `agent` & `proxy`!

Reply:
So entity type is actually sort of confusing.

An agent entity is a reserved word and should only be used by sensu agents.

I can’t actually find anywhere in the code that a proxy entity type means anything via  simple search, and I can’t think of anything off the top of my head.

People can put whatever value they want here—but if they put agent, then it invokes very specific functionality on the backend.

Honestly. We should rethink this field. The agent entity type is a very opaque feature.

Reply:
I don’t think there are actually allowed values here, but agent and proxy are reserved. Agent for sensu-agent processes and proxy for proxy entities.

Reply:
You are totally right, it's actually not enforced, my bad!

Reply:
Should I add `agent` as a default value?

Reply:
AFAIK the agent will set that value for its own entity and use `proxy` for result submitted through the socket & proxy checks, but otherwise it's not enforced by the API/backend. As Greg mentioned in its deleted comment, maybe the role of this attribute is not clear enough?

Reply:
Ok that makes sense. I'll clarify that this value is set by the agent/proxy.